tests: drivers: : ethernet: fails to link for sam_v71_xult and sam_v71b_xult

**Describe the bug**
The daily build fails to link for this test / platform combination.

**To Reproduce**
```
twister -i -p sam_v71_xult -s tests/drivers/build_all/ethernet/net.ethernet.build
twister -i -p sam_v71b_xult -s tests/drivers/build_all/ethernet/net.ethernet.build
```

**Expected behavior**
The daily build should not fail.

**Impact**
Showstopper for anyone using ethernet on these platforms.

**Logs and console output**
```
/opt/toolchains/zephyr-sdk-0.13.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/spi/libdrivers__spi.a(spi_sam.c.obj):(.rodata.__compound_literal.0+0x0): undefined reference to `__device_dts_ord_27'
/opt/toolchains/zephyr-sdk-0.13.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/spi/libdrivers__spi.a(spi_sam.c.obj):(.rodata.__compound_literal.0+0x8): undefined reference to `__device_dts_ord_27'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**
 - OS: CI
 - Toolchain: Zephyr SDK 0.13.2
 - Commit SHA: 7f5eb921dc8bb1b7014886986bbffdf111b68b15

**Additional context**
Daily build failure:
https://github.com/zephyrproject-rtos/zephyr/runs/4676614702?check_suite_focus=true

--- a/drivers/gpio/Kconfig.sam
+++ b/drivers/gpio/Kconfig.sam
@@ -4,16 +4,21 @@
 # Copyright (c) 2020 Gerson Fernando Budke <nandojve@gmail.com>
 # SPDX-License-Identifier: Apache-2.0
 
+# Workaround for not being able to have commas in macro arguments
+DT_COMPAT_ATMEL_SAM_GPIO := atmel,sam-gpio
+
 config GPIO_SAM
 	bool "Atmel SAM GPIO (PORT) driver"
-	default y
+	default y if $(dt_compat_enabled,$(DT_COMPAT_ATMEL_SAM_GPIO))
 	depends on SOC_FAMILY_SAM && !SOC_SERIES_SAM4L
+	depends on GPIO
 	help
 	  Enable support for the Atmel SAM 'PORT' GPIO controllers.
 
 config GPIO_SAM4L
 	bool "Atmel SAM4L GPIO (PORT) driver"
-	default y
+	default y if $(dt_compat_enabled,$(DT_COMPAT_ATMEL_SAM_GPIO))
 	depends on SOC_SERIES_SAM4L
+	depends on GPIO
 	help
 	  Enable support for the Atmel SAM4L 'PORT' GPIO controllers.

--- a/drivers/spi/Kconfig.sam
+++ b/drivers/spi/Kconfig.sam
@@ -4,9 +4,14 @@
 # Copyright (c) 2018 qianfan Zhao
 # SPDX-License-Identifier: Apache-2.0
 
+# Workaround for not being able to have commas in macro arguments
+DT_COMPAT_ATMEL_SAM_SPI := atmel,sam-spi
+
 config SPI_SAM
 	bool "Atmel SAM series SPI driver"
-	default y
+	default $(dt_compat_enabled,$(DT_COMPAT_ATMEL_SAM_SPI))
 	depends on SOC_FAMILY_SAM
+	depends on SPI
+	select GPIO
 	help
 	  Enable support for the SAM SPI driver.

--- a/soc/arm/atmel_sam/samv71/Kconfig.defconfig.series
+++ b/soc/arm/atmel_sam/samv71/Kconfig.defconfig.series
@@ -46,10 +46,6 @@ config DMA_SAM_XDMAC
 	default y
 	depends on DMA
 
-config GPIO_SAM
-	default y
-	depends on GPIO
-
 config ADC_SAM_AFEC
 	default y
 	depends on ADC
@@ -61,10 +57,6 @@ config I2C_SAM_TWIHS
 config I2S_SAM_SSC
 	default y
 	depends on I2S
-
-config SPI_SAM
-	default y
-	depends on SPI
 
 config USB_DC_SAM_USBHS
 	default y

--- a/tests/drivers/spi/spi_loopback/boards/sam_v71_xult.conf
+++ b/tests/drivers/spi/spi_loopback/boards/sam_v71_xult.conf
@@ -1,1 +1,0 @@
-CONFIG_GPIO=y

--- a/tests/drivers/spi/spi_loopback/boards/sam_v71b_xult.conf
+++ b/tests/drivers/spi/spi_loopback/boards/sam_v71b_xult.conf
@@ -1,1 +1,0 @@
-CONFIG_GPIO=y

